src/constants/alignment.js

```diff
@@ -0,0 +1,32 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+// fraction of some size to get to a named position
+module.exports = {
+    // from bottom left: this is the origin of our paper-reference
+    // positioning system
+    FROM_BL: {
```

Comment:
Ah, wasn't even thinking of that but good point. There are tons of other places we could make use of these (potentially adding `FROM_BR` and `FROM_TR`...) to make things clearer and more efficient. see eg `ag middle:` in the src directory. I guess perhaps all of these could also allow `[0, 1]`.